### PR TITLE
Fix lateral panic when using motion to pass through params

### PR DESCRIPTION
**Pull request by zxuejing**
GPDB does not support passing params by a motion.
If we pass params by a motion. It will cause panic.
```
create table ttt(tc1 varchar(10)) distributed randomly;
create table ttt1(tc2 varchar(10)) distributed randomly;
insert into ttt values('sdfs');
insert into ttt1 values('sdfs');
  
explain (costs off)
select 
   ttt.*,  t.t1 
from 
  ttt 
  left join lateral (
    select 
      string_agg(distinct tc2, ';') as t1 
    from 
      ttt1 
    where 
      ttt.tc1=ttt1.tc2
 ) t on true;
  
  --------------------------------------------------------------------------------------
--------
 Nested Loop Left Join  (cost=10000000001.06..10000000002.12 rows=4 width=37)
   ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..1.03 rows=1 width=5)
         ->  Seq Scan on ttt  (cost=0.00..1.01 rows=1 width=5)
   ->  Materialize  (cost=1.06..1.08 rows=1 width=32)
         ->  Aggregate  (cost=1.06..1.07 rows=1 width=32)
               ->  Gather Motion 3:1  (slice2; segments: 3)  (cost=0.00..1.05 rows=1 w
idth=5)
                     ->  Seq Scan on ttt1  (cost=0.00..1.01 rows=1 width=5)
                           Filter: ((ttt.tc1)::text = (tc2)::text)
 Optimizer: Postgres query optimizer
```

Add a walker in create_nestloop_plan to check whether extParam of a motion in the rightPlan
contains any NestLoopParam. 
In that case, we throw an error instead of panic.

### CODE REVIEW

**Comment by kainwen** [comment]
Thanks for your work! Need more thoughts on this PR.
Can we add case of the issue https://github.com/greenplum-db/gpdb/issues/10013 in this PR?

**Comment by zxuejing** [comment]
> Thanks for your work! Need more thoughts on this PR. Can we add case of the issue #10013 in this PR?

Ok, I will add testcases in #10013 to this pr.

File: src/test/regress/expected/gporca.out [hunk 1]
```diff
@@ -12633,29 +12611,7 @@ select *
 from tcorr1 out
 where out.b in (select coalesce(tcorr2.a, 99)
                 from tcorr1 left outer join tcorr2 on tcorr1.a=tcorr2.a+out.a);
-                                                         QUERY PLAN
```
Comment (kainwen) [inline src/test/regress/expected/gporca.out:12636]: The removed test case result is interesting. It pull the any sublink to a lateral join. Postgres never do such kind of pullup. Greenplum officially does not support lateral join in 6X yet.

Need spend time on this.

Other parts of code seems fine.

File: src/backend/optimizer/plan/createplan.c [hunk 2]
```diff
@@ -3281,6 +3286,13 @@ create_nestloop_plan(PlannerInfo *root,
 							  inner_plan,
 							  best_path->jointype);
 
+	/*
+	 * Check whether pass lateral params by a motion.
+	 */
+	if (!bms_is_empty(bmsNestParams))
+	{
```
Comment (kainwen) [inline src/backend/optimizer/plan/createplan.c:3293]: single line body might not need `{}`.

**Review by kainwen — COMMENTED** [review]
The implementation LGTM. Leave some comments.

**Review by kainwen — APPROVED** [review]

File: src/test/regress/expected/gporca.out (hunk 1 above)
Comment (zxuejing) [reply src/test/regress/expected/gporca.out:12636]: I will create an issue to record this problem, will investigate why pull the any sublink to a lateral join.

File: src/backend/cdb/cdbmutate.c [hunk 3]
```diff
@@ -3670,3 +3670,54 @@ contains_outer_params(Node *node, void *context)
...
+	if (IsA(node, Motion))
+	{
+		Plan * plan = (Plan *) node;
+		Bitmapset  *finalExtParam;
+		if (!bms_is_empty(plan->extParam))
+		{
+			finalExtParam = bms_intersect(plan->extParam, motionWithParamcontext->nestLoopParams);
+			if (!bms_is_empty(finalExtParam))
+			{
+				elog(ERROR, "There is a param bellow Motion.");
```
Comment (kmacoskey) [inline src/backend/cdb/cdbmutate.c:3704]: The error message could be expanded. A suggestion would be:

"Passing parameters across motion is not supported"

**Review by kmacoskey — APPROVED** [review]

File: src/backend/cdb/cdbmutate.c (hunk 3 above)
Comment (zxuejing) [reply src/backend/cdb/cdbmutate.c:3704]: thanks！

**Pull request merged**